### PR TITLE
Swissprot bugfix

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request addresses issue #2417  . Travis, bring it on.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by mdehoon** [comment]
travis error is due to an unrelated HTML build error

**Comment by fredricj** [comment]
This fails on entry P93784.txt with:
Traceback (most recent call last):
  File "split_archive.py", line 35, in <module>
    record = SwissProt.read(buf)
  File "/opt/biopython-venv/lib/python3.6/site-packages/Bio/SwissProt/__init__.py", line 251, in read
    record = _read(handle)
  File "/opt/biopython-venv/lib/python3.6/site-packages/Bio/SwissProt/__init__.py", line 349, in _read
    _read_ft(record, line)
  File "/opt/biopython-venv/lib/python3.6/site-packages/Bio/SwissProt/__init__.py", line 799, in _read_ft
    first_seq, second_seq = value.split(" -> ")
UnboundLocalError: local variable 'value' referenced before assignment

**Comment by fredricj** [comment]
I meant P48349 and not P93784 as I originally wrote. This PR does not handle multiline FT properly as is.

**Comment by fredricj** [comment]
It also breaks with P0CK95 where a note description begins a second line with a /

**Comment by mdehoon** [comment]
@fredricj  I believe for P48349 the parser is working correctly now. Can you prepare a test case for P0CK95 ? (I mean a unit test we can add to `test_SwissProt.py`)

**Comment by fredricj** [comment]
Yes, P48349 works fine now. I have created a PR you should be able to merge into this. Im not certain on the spacing when joining the lines but used same as in the XML version

**Comment by fredricj** [comment]
With the latest changes P0CK95 also works, as well as the whole uniprot_sprot.dat for 2019_11 although  if it does it correctly for all entries I cant really say

**Comment by mdehoon** [comment]
@fredricj  Thanks! Let's see if @peterjc  has any additional comments before we merge.

File: Bio/SeqFeature.py [hunk 1]
```diff
@@ -47,6 +47,7 @@
  - BeforePosition - Specify the position as being found before some base.
  - AfterPosition - Specify the position as being found after some base.
  - OneOfPosition - Specify a position where the location can be multiple positions.
+ - UncertainPosition - Specify a specific position which is uncertain.
```
Comment (peterjc) [inline Bio/SeqFeature.py]: Nice catch, an oversight in the module documentation

File: Bio/SwissProt/__init__.py [hunk 2]
```diff
@@ -16,6 +16,18 @@
 
 """
 
+# flake8: noqa
```
Comment (peterjc) [inline Bio/SwissProt/__init__.py]: Why is this being used? I think it turns off flake8 for the entire file which is a very bad idea.

File: Bio/SwissProt/__init__.py [hunk 3]
```diff
@@ -136,6 +148,89 @@ def __init__(self):
...
+                    sequence; the location is stored as an instance of
+                    FeatureLocation, defined in Bio.SeqFeature, with the ref
+                    attribute set to the isoform ID referring to the canonical
+                    or isoform sequence on which the feature is defined
+     - id           unique and stable identifier (FTId), only provided for
+                    features belonging to the types CARBOHYD, CHAIN, PEPTIDE,
+                    PROPEP, VARIANT, or VAR_SEQ
+     - type         indicates the type of feature, as defined by the UniProt
+                    Knowledgebase documentation:
+                        ACT_SITE: amino acid(s) involved in the activity of an
```
Comment (peterjc) [inline Bio/SwissProt/__init__.py]: This block of intended text is probably problematic as RST - have you checked?

**Comment by peterjc** [comment]
I would like to see a matched set of examples like ``Tests/UniProt/multi_ex.xml`` etc (or perhaps update that existing test set to the new format) for validation of SwissProt vs UniProt XML vs FASTA etc. Does that seem too much work?

File: Bio/SwissProt/__init__.py (hunk 3 above)
Comment (peterjc) [reply Bio/SwissProt/__init__.py]: This is almost certainly why the RST documentation build is failing on TravisCI

**Comment by mdehoon** [comment]
fixed now, thanks for the explanation

**Comment by lizhang12** [comment]
File "/home/lizhang12/.local/lib/python3.6/site-packages/Bio/SwissProt/__init__.py", line 798, in _read_ft
    first_seq, second_seq = value.split(" -> ")
UnboundLocalError: local variable 'value' referenced before assignment

**Comment by peterjc** [comment]
@lizhang12 could you clarify which version you are testing (e.g. a release, or the specific commit if from github) as that doesn't seem to match the current latest code on github.

**Comment by lizhang12** [comment]
> @lizhang12 could you clarify which version you are testing (e.g. a release, or the specific commit if from github) as that doesn't seem to match the current latest code on github.

Thank you so much for the reminder! I have updated to the latest code and it works well now!

**Comment by peterjc** [comment]
Excellent, pleased to hear that @lizhang12

**Pull request merged**